src/serializer/Referentializer.js

```diff
@@ -161,8 +164,10 @@ export class Referentializer {
   _getReferentializationScope(residualBinding: ResidualFunctionBinding): ReferentializationScope {
     if (residualBinding.potentialReferentializationScopes.has("GLOBAL")) return "GLOBAL";
     if (residualBinding.potentialReferentializationScopes.size > 1) {
-      // TODO #2428: Revisit for nested optimized functions.
-      return "GLOBAL";
+      invariant(this.helperState !== undefined);
+      let scopes = ((residualBinding.potentialReferentializationScopes: any): Set<Scope>);
```

Comment:
`Scope` and `ReferentializationScope` are different. Why is it a good idea to do this casting here?